src/test/lib/QuicTest.cpp

```diff
@@ -388,6 +388,69 @@ struct PrivateTransportHelper : QUIC_PRIVATE_TRANSPORT_PARAMETER
     }
 };
 
+struct RandomLossHelper
+{
+    static uint8_t LossPercentage;
+    QUIC_TEST_DATAPATH_FUNC_TABLE DataPathFuncTable;
+    RandomLossHelper(uint8_t _LossPercentage) {
+        LossPercentage = _LossPercentage;
+        DataPathFuncTable.Receive = ReceiveCallback;
+        DataPathFuncTable.Send = SendCallback;
+        if (LossPercentage != 0) {
+            QUIC_TEST_DATAPATH_FUNC_TABLE* Value = &DataPathFuncTable;
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_LEVEL_GLOBAL,
+                    QUIC_PARAM_GLOBAL_TEST_DATAPATH_FUNC_TABLE,
+                    sizeof(Value),
+                    &Value));
+        }
+    }
+    ~RandomLossHelper() {
+        if (LossPercentage != 0) {
+            QUIC_TEST_DATAPATH_FUNC_TABLE* Value = nullptr;
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->SetParam(
+                    nullptr,
+                    QUIC_PARAM_LEVEL_GLOBAL,
+                    QUIC_PARAM_GLOBAL_TEST_DATAPATH_FUNC_TABLE,
+                    sizeof(Value),
+                    &Value));
+        }
+    }
+    static
+    _IRQL_requires_max_(DISPATCH_LEVEL)
+    void
+    QUIC_API
+    ReceiveCallback(
+        _Inout_ struct QUIC_RECV_DATAGRAM* DatagramChain
+        )
+    {
+        uint8_t RandomValue;
+        while (DatagramChain) {
+            QuicRandom(sizeof(RandomValue), &RandomValue);
+            DatagramChain->Drop = (RandomValue % 100) >= LossPercentage;

```

Comment:
If it's convenient to make this change right now, then please do so: we should have a loss divisor rather than a percentage, since 1% is already a large loss rate. I.e.: the parameter is "Divisor," and if (rand()/Divisor == 0) then the packet is dropped. This means for 1% loss you'd set Divisor=100 and for .1% loss Divisor=1000, etc.

Reply:
If it's not convenient, then no problem.

Reply:
I think I'm going to leave this as for now. Feel free to make a PR after I merge this to implement your suggestion.